flink-table-store-core/src/main/java/org/apache/flink/table/store/file/operation/AbstractFileStoreScan.java

```diff
@@ -87,6 +89,7 @@ public AbstractFileStoreScan(
         this.numOfBuckets = numOfBuckets;
         this.checkNumOfBuckets = checkNumOfBuckets;
         this.changelogProducer = changelogProducer;
+        this.readCompacted = readCompacted;
```

Comment:
What happens if user set changelog producer to `INPUT`, and at the same time use `readCompacted`? It seems that we'll scan nothing, which is unfriendly to the user.